src/radical/pilot/pmgr/launching/default.py

```diff
@@ -884,6 +884,7 @@ def _prepare_pilot(self, resource, rcfg, pilot, expand):
         # pilot description and resource configuration
         number_cores    = pilot['description']['cores']
         number_gpus     = pilot['description']['gpus']
+        required_memory = pilot['description']['memory']  # per node
```

Comment:
I think this should not be per node, but total memory, to be in sync with the other attributes.  Also, the user is not even supposed to know in how many nodes the given number of cores and GPUs results.

Reply:
Thus we assume that user set the total amount memory? which later we suppose to divide by number of nodes? (that comment I've put for us, since later this parameter translates into `--mem`, which is "the real memory required per node")